google-cloud-core/src/main/java/com/google/cloud/Timestamp.java

```diff
@@ -119,7 +119,13 @@ public static Timestamp now() {
    * @throws IllegalArgumentException if the timestamp is outside the representable range
    */
   public static Timestamp of(java.sql.Timestamp timestamp) {
-    return ofTimeSecondsAndNanos(timestamp.getTime() / 1000, timestamp.getNanos());
+    // TODO: replace with Math.floorDiv when we drop Java 7 support
+    long secs = timestamp.getTime() / 1000;
+    int nanos = timestamp.getNanos();
+    if (secs < 0) {
+      --secs;
```

Comment:
This feels like a hack. Why does this happen in the first place? Is this expected behavior?

Either way, it should probably have comments in the code.

Reply:
Added a comment about why the hack is in there and a TODO for switching to `Math.floorDiv` when we drop Java 7 support